sc/src/main/java/uk/ac/ebi/atlas/search/GeneSearchService.java

```diff
@@ -84,7 +85,8 @@ public GeneSearchService(GeneSearchDao geneSearchDao, TsnePlotSettingsService ts
                     geneSearchDao.fetchClusterIdsWithPreferredKAndMinPForExperimentAccession(
                             geneId,
                             experimentAccession,
-                            preferredK.orElse(null));
+                            // If there’s no preferred k we use 0, which won’t match any row
+                            preferredK.orElse(0));
```

Comment:
I think neither `null` nor `0` are elegant approaches here. I think separate queries would make things a bit clearer and more explicit... We could improve upon this at a later date of course.

Reply:
Agreed, in the interest of time I’ll add a ticket to Pivotal and leave it as it is if you don’t have anything against it. @lingyun1010?